benches/r1cs.rs

```diff
@@ -156,7 +156,7 @@ impl KShuffleGadget {
             .unzip();
 
         Self::fill_cs(&mut prover, input_vars, output_vars)?;
-        let proof = prover.prove(&bp_gens)?;
+        let proof = prover.prove(bp_gens, |capacity, gens| gens.increase_capacity(capacity))?;
```

Comment:
maybe instead of sending generators into `prove`, and then getting them into closure, instead simply provide the reference from the closure (not sure if borrowck would love this):

```rust
let gens = BulletproofGens::new(128, 1);
...
let proof = prover.prove(|capacity| gens.increase_capacity(capacity); &gens )?;
```

?

Reply:
And the closure must return `Result<&BulletproofGens, R1CSError>` because the prover may decide to return Err if capacity is too high instead of blinding increasing it.

Reply:
Having the closure provide the reference makes the borrow checker unhappy 😢 says it can't infer the appropriate lifetime?

Reply:
The prover can provide a closure that doesn't resize the generators (which would result in a failure from insufficient capacity). Do we want to also have the closure return `Result<(), R1CSError>`? It feels maybe unnecessary to me since a no-op closure will have the same overall effect